### PR TITLE
GH-42100: [C++][Parquet] ParquetFilePrinter::JSONPrint print length of FLBA

### PR DESCRIPTION
**Pull request by mapleFU**
### Rationale for this change

Print FLBA length in `ParquetFilePrinter::JSONPrint`

### What changes are included in this PR?

Print FLBA length in `ParquetFilePrinter::JSONPrint`

### Are these changes tested?

no need

### Are there any user-facing changes?

no

* GitHub Issue: #42100

### CODE REVIEW

**Comment by mapleFU** [comment]
cc @pitrou @wgtmac

**Comment by pitrou** [comment]
I'm not sure this is minor, do you want to open an issue?

**Comment by pitrou** [comment]
Also, do we want to add basic tests for ParquetFilePrinter at some point?

**Comment by mapleFU** [comment]
Emmm I use this as debug tool and I found it doesn't contains FLBA length, it's just debug tool, so I think adding test for this would make `Printer` weird to maintain...

**Comment by pitrou** [comment]
Well, wouldn't it be useful to check that the debugging tool doesn't regress?
I'm not insisting on tests, just to note that the "not needed" in your PR description seems a bit cavalier.

**Comment by pitrou** [comment]
cc @wgtmac

**Comment by wgtmac** [comment]
I agree that it might be useful to add a test like this to make sure it is not broken in the future: https://github.com/apache/arrow/blob/main/cpp/src/parquet/reader_test.cc#L1080-L1144

**Comment by mapleFU** [comment]
Ok, I can add one like that

**Comment by mapleFU** [comment]
@wgtmac @pitrou what do you think adding test for inspect `alltypes_plain.parquet`? The recent `alltypes_plain.parquet` might introduce more new features in Parquet ( like column index, bloom filter, etc )

**Comment by pitrou** [comment]
@mapleFU Sounds good to me.

**Comment by mapleFU** [comment]
I came cross a problem:

```
"ColumnChunks": [
          {"Id": "0", "Values": "1000", "StatsSet": "True", "Stats": {"NumNulls": "105", "Max": "�", "Min": "" },
           "Compression": "UNCOMPRESSED", "Encodings": "PLAIN", "UncompressedSize": "3839", "CompressedSize": "3839", ColumnIndex {"offset": "3843", "length": "131"}", OffsetIndex {"offset": "3974", "length": "101"}" }
        ]
```

Min-Max stats for FLBA would be a bit weird 🤔

**Comment by pitrou** [comment]
@mapleFU We can fix that in another PR.

File: cpp/src/parquet/reader_test.cc [hunk 1]
```diff
@@ -1077,7 +1077,24 @@ Column 1
   ASSERT_EQ(header_output + dump_output, ss_dump.str());
 }
 
-TEST(TestJSONWithLocalFile, JSONOutput) {
+class TestJSONWithLocalFile : public ::testing::Test {
+ public:
+  static std::string readFromLocalFile(std::string_view localFileName) {
```
Comment (pitrou) [inline cpp/src/parquet/reader_test.cc:1082]: Nit: coding style
```suggestion
  static std::string ReadFromLocalFile(std::string_view file_path) {
```

Comment (mapleFU) [reply cpp/src/parquet/reader_test.cc]: done

**Review by pitrou — APPROVED** [review]
+1, feel free to merge

**Comment by mapleFU** [comment]
https://github.com/apache/arrow/issues/42104 Create issue here

**Pull request merged**